src/main/groovy/com/github/dispader/manifesto/ManifestoPlugin.groovy

```diff
@@ -10,67 +10,23 @@ class ManifestoPlugin implements Plugin<Project> {
-
-                    attributes('Specification-Title': project.name)
-
-                    if ( project.manifesto?.vendor ) {
-                        attributes('Specification-Vendor': "${project.manifesto.vendor}")
-                    } else {
-                        project.logger.warn "warning: Manifest Specification-Vendor not set."
-                    }
-
-                    if ( version.specification ) {
-                        attributes('Specification-Version': version.specification)
-                    } else if ( project.version ) {
-                        attributes('Specification-Version': "${project.version}")
-                    } else {
-                        project.logger.warn "warning: Manifest Specification-Version not set."
-                    }
-
-                    attributes('Implementation-Title': project.name)
-
-                    if ( project.manifesto?.vendor ) {
-                        attributes('Implementation-Vendor': "${project.manifesto.vendor}")
-                    } else {
-                        project.logger.warn "warning: Manifest Implementation-Vendor not set."
-                    }
-
-                    if ( project.manifesto?.vendor_id ) {
-                        attributes('Implementation-Vendor-Id': "${project.manifesto.vendor_id}")
-                    } else {
-                        project.logger.warn "warning: Manifest Implementation-Vendor-Id not set."
-                    }
-
-                    if ( project.manifesto?.url ) {
-                        attributes('Implementation-URL': "${project.manifesto.url}")
-                    } else {
-                        project.logger.warn "warning: Manifest Implementation-URL not set."
-                    }
-
-                    if ( version.implementation ) {
-                        attributes('Implementation-Version': version.implementation)
-                    } else if ( project.version ) {
-                        attributes('Implementation-Version': "${project.version}")
-                    } else {
-                        project.logger.warn "warning: Manifest Implementation-Version not set."
-                    }
-
-                    attributes('Implementation-Timestamp': new Date())
+                    attributes << [
+                        'Specification-Title'     : project.name,
+                        'Specification-Vendor'    : project.manifesto.vendor,
+                        'Specification-Version'   : version.specification,
+                        'Implementation-Title'    : project.name,
+                        'Implementation-Vendor'   : project.manifesto.vendor,
+                        'Implementation-Vendor-Id': project.manifesto.vendor_id,
+                        'Implementation-URL'      : project.manifesto.url,
+                        'Implementation-Version'  : version.implementation,
+                        'Implementation-Timestamp': new Date()
+                    ]
```

Comment:
I like this a lot, and think it's way less chatty with the log messages, and I'm loving the syntax here.  The only thing that gives me pause is whether things like `project.name` will come back as empty or `null`.  If we're not populating manifests with `null`, I'm way on board with this.